src/prefect/client/orchestration.py

```diff
@@ -283,12 +284,18 @@ def __init__(
         httpx_settings.setdefault("headers", {})
 
         if PREFECT_API_TLS_INSECURE_SKIP_VERIFY:
-            httpx_settings.setdefault("verify", False)
+            # Create an unverified context for insecure connections
+            ctx = ssl.create_default_context()
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
+            httpx_settings.setdefault("verify", ctx)
         else:
             cert_file = PREFECT_API_SSL_CERT_FILE.value()
             if not cert_file:
                 cert_file = certifi.where()
-            httpx_settings.setdefault("verify", cert_file)
+            # Create a verified context with the certificate file
+            ctx = ssl.create_default_context(cafile=cert_file)
+            httpx_settings.setdefault("verify", ctx)
```

Comment:
You are repeating a few things in both branches that could be simplified out:

```python
        ctx = ssl.create_default_context()
        if PREFECT_API_TLS_INSECURE_SKIP_VERIFY:
            # Update the context to not verify for insecure connections
            ctx.check_hostname = False
            ctx.verify_mode = ssl.CERT_NONE
        else:
            cert_file = PREFECT_API_SSL_CERT_FILE.value()
            if not cert_file:
                cert_file = certifi.where()
            # Update the context with the certificate file
            ctx.load_verify_locations(cafile=cert_file)
        httpx_settings.setdefault("verify", ctx)
```

Reply:
hi @mjpieters! you are correct, thanks for pointing this out! in this draft PR i was exploring what fixes would be necessary

Reply:
Ah, of course, sorry if I jumped the gun a bit here.

Reply:
@mjpieters no worries! if you're interested in following along, we're working through the same upstream issue's impact on our tests [here](https://github.com/PrefectHQ/prefect/pull/16168)